src/__tests__/SSEConnection-test.js

```diff
@@ -0,0 +1,199 @@
+import { sources } from "eventsourcemock";
+import puppeteer from "puppeteer";
+import express from "express";
+
+import SSEConnection from "../SSEConnection";
+import ConnectionEvent from "../ConnectionEvent";
+
+const url = "https://mesosphere.com";
+const wait = ms => new Promise(resolve => setTimeout(resolve, ms));
+
+describe("Server Sent Event Connection", () => {
+  describe("API", () => {
+    let connection;
+    beforeEach(() => {
+      connection = new SSEConnection(url);
+    });
+
+    describe("#constructor", () => {
+      it("fails without an url", () => {
+        expect(() => {
+          new SSEConnection();
+        }).toThrowError();
+      });
+    });
+
+    describe("#open", () => {
+      it("opens a connection", () => {
+        const onOpenEventMock = jest.fn();
+        expect(connection.state).toBe(SSEConnection.INIT);
+        connection.addListener(ConnectionEvent.OPEN, onOpenEventMock);
+
+        connection.open();
+        sources[url].emitOpen();
+
+        expect(sources[url].readyState).toBe(1);
+        expect(connection.state).toBe(SSEConnection.OPEN);
+        expect(onOpenEventMock).toHaveBeenCalled();
+      });
+
+      it("uses the config options", () => {
+        connection = new SSEConnection(url, {
+          withCredentials: true
+        });
+        connection.open();
+
+        expect(sources[url].withCredentials).toBe(true);
+      });
+    });
+
+    describe("on message", () => {
+      const msg = "My message from the server";
+      beforeEach(() => {
+        connection.open();
+        sources[url].emitOpen();
+      });
+
+      it("invokes listener", () => {
+        const onDataEventMock = jest.fn();
+        connection.addListener(ConnectionEvent.DATA, onDataEventMock);
+
+        sources[url].emitMessage(msg);
+
+        expect(onDataEventMock).toHaveBeenCalled();
+      });
+
+      it("adds it to the data attribute", () => {
+        sources[url].emitMessage(msg);
+        expect(connection.data[0]).toBe(msg);
```

Comment:
as already stated, please add some kind of getter for this. We dont want anything external to be able to mutate this data.